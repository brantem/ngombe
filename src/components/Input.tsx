import { forwardRef } from 'react';

import { cn } from 'lib/helpers';

type InputProps = React.ComponentPropsWithoutRef<'input'> & {
  label?: string;
  invalid?: boolean;
};

const Input = forwardRef<HTMLInputElement, InputProps>(function Input({ label, className, invalid, ...props }, ref) {
  return (
    <label className="flex flex-col">
      {label && <span className="mb-1 text-neutral-500">{label}</span>}
      <input
        ref={ref}
        className={cn(
          'text-2xl font-extrabold text-neutral-700 outline-none rounded-xl h-12 px-3 border-2 border-neutral-200 transition-all',
          invalid && 'border-red-500',
          className,
        )}
        {...props}
      />
    </label>
  );
});

export default Input;
