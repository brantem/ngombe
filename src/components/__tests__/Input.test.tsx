import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Input from 'components/Input';

test('Input', () => {
  const { container } = render(<Input label="Label" />);
  expect(container).toMatchSnapshot();
});
