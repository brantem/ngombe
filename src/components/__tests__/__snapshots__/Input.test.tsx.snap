// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input 1`] = `
<div>
  <label
    class="flex flex-col"
  >
    <span
      class="mb-1 text-neutral-500"
    >
      Label
    </span>
    <input
      class="text-2xl font-extrabold text-neutral-700 outline-none rounded-xl h-12 px-3 border-2 border-neutral-200 transition-all"
    />
  </label>
</div>
`;
